_posts/2022-01-10-reassign-partitions.md

```diff
@@ -0,0 +1,363 @@
+So the answer to this is the topics that have their partitions assigned to broker `<CLUSTER-NAME>-kafka-3` and `<CLUSTER-NAME>-kafka-4` needs reassignment, and these topics partition should move to the remaining 3 brokers nodes in our cluster.
+
+To check the partitions details of a certain topic, we can use the `kafka-topics.sh` tool. We can run the following command:
+
+```sh
+bin/kafka-topics.sh --describe --topic my-topic-1 --bootstrap-server <CLUSTER-NAME>-kafka-bootstrap:9093 --command-config /tmp/config.properties
+```
+which will give us the following output:
+
+```shell
+Topic: my-topic-1       TopicId: bW1J-3OESJ2MF6buaLkkkQ PartitionCount: 10      ReplicationFactor: 3    Configs: min.insync.replicas=2,segment.bytes=1073741824,retention.ms=7200000,message.format.version=3.0-IV1
+        Topic: my-topic-1       Partition: 0    Leader: 2       Replicas: 2,3,0 Isr: 2,3,0
+        Topic: my-topic-1       Partition: 1    Leader: 3       Replicas: 3,0,1 Isr: 3,0,1
+        Topic: my-topic-1       Partition: 2    Leader: 0       Replicas: 0,1,4 Isr: 0,1,4
+        Topic: my-topic-1       Partition: 3    Leader: 1       Replicas: 1,4,2 Isr: 1,4,2
+        Topic: my-topic-1       Partition: 4    Leader: 4       Replicas: 4,2,3 Isr: 4,2,3
+        Topic: my-topic-1       Partition: 5    Leader: 2       Replicas: 2,0,1 Isr: 2,0,1
+        Topic: my-topic-1       Partition: 6    Leader: 3       Replicas: 3,1,4 Isr: 3,1,4
+        Topic: my-topic-1       Partition: 7    Leader: 0       Replicas: 0,4,2 Isr: 0,4,2
+        Topic: my-topic-1       Partition: 8    Leader: 1       Replicas: 1,2,3 Isr: 1,2,3
+        Topic: my-topic-1       Partition: 9    Leader: 4       Replicas: 4,3,0 Isr: 4,3,0
+```
+In the same way you can get the details for the other topic`my-topic` also.
+
+```sh
+Topic: my-topic TopicId: ERoAsjHHTIKFRPoo3h_ZZg PartitionCount: 10      ReplicationFactor: 3    Configs: min.insync.replicas=2,segment.bytes=1073741824,retention.ms=7200000,message.format.version=3.0-IV1
+        Topic: my-topic Partition: 0    Leader: 1       Replicas: 1,2,0 Isr: 0,1,2
+        Topic: my-topic Partition: 1    Leader: 2       Replicas: 2,0,1 Isr: 2,1,0
+        Topic: my-topic Partition: 2    Leader: 0       Replicas: 0,1,2 Isr: 0,1,2
+        Topic: my-topic Partition: 3    Leader: 0       Replicas: 1,0,2 Isr: 0,2,1
+        Topic: my-topic Partition: 4    Leader: 1       Replicas: 2,1,0 Isr: 1,0,2
+        Topic: my-topic Partition: 5    Leader: 0       Replicas: 0,2,1 Isr: 1,2,0
+        Topic: my-topic Partition: 6    Leader: 2       Replicas: 1,2,0 Isr: 0,1,2
+        Topic: my-topic Partition: 7    Leader: 2       Replicas: 2,0,1 Isr: 2,0,1
+        Topic: my-topic Partition: 8    Leader: 0       Replicas: 0,1,2 Isr: 0,1,2
+        Topic: my-topic Partition: 9    Leader: 1       Replicas: 1,0,2 Isr: 1,0,2
+```
+
+From the above outputs, we got to know that `my-topic-1` needs reassignment, Now we can create the topics.json file easily.
```

Comment:
fyi this is the point where I saw `my-topic-1` and wondered if it was a topic or a partition, prompting my previous comment.